File: Tools/factors.py


def factors(n):
    factors = []
    for i in range(1, n+1):
        if n % i == 0:
            factors.append(i)
    return factors

def factorial(n):
    if n == 0 or n == 1:
        return 1
    else:
        return n * factorial(n-1)

if __name__ == '__main__':
    nums = [1, 3, 6, 10, 15, 21, 28]
    for i in nums:
        print(factors(i))

    # print(factorial(15))

File: util/Factors.java
package util;

import java.util.ArrayList;
import java.util.List;

public class Factors {

    static List<Integer> factors(int number) {
        List<Integer> list = new ArrayList<>();
        for (int i = 1; i <= number; i++) {
            if (number % i == 0)
                list.add(i);
        }

        return list;
    }
    static int factorial(int number) {
        if (number == 0 || number == 1) {
            return 1;
        }

        return number * factorial(number - 1);
    }
}
